jsonpb/jsonpb.go

```diff
@@ -637,15 +637,18 @@ func UnmarshalString(str string, pb proto.Message) error {
 func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMessage, prop *proto.Properties) error {
 	targetType := target.Type()
 
-	// Allocate memory for pointer fields.
-	if targetType.Kind() == reflect.Ptr {
+	switch targetType.Kind() {
+	case reflect.Map, reflect.Ptr, reflect.Slice:
```

Comment:
Let's not combine the logic for map and slice here.  It's better to just leave this for pointer type alone.

Suggest adding the following logic instead ...

```
if string(inputValue) == "null" {
    return nil
}
```

after original file lines 917 for handling slice
```
if targetType.Kind() == reflect.Slice...
```
and after line 933
```
if targetType.Kind() == reflect.Map {
```